test/test.hs

```diff
@@ -282,7 +282,12 @@ tests = testGroup "hevm"
     ]
   , testGroup "SimplifierUnitTests"
     -- common overflow cases that the simplifier was getting wrong
-    [ testCase "writeWord-overflow" $ do
+    [ testCase "bufLength-simp" $ do
```

Comment:
Would adding tests for more complicated buffer expressions (the involve writes perhaps to abstract buffers) make sense?

Reply:
Probably :) I wanted to keep this PR short and sweet. I didn't want to dig myself into `bufLength` -- it may be perfect, it may be possible to improve. I just wanted to make sure it's actually working and used :smile: 

Should we write a few more test cases? Perhaps that could be an issue I can tackle in another PR. I don't want to delay this PR too long. What do you think?